packages/cascara/src/modules/DataCheckbox/DataCheckbox.js

```diff
@@ -31,26 +32,37 @@ const DataCheckbox = ({
   const { isEditing, formMethods } = useContext(ModuleContext);
   const checkbox = useCheckboxState({ state: Boolean(value) });
 
+  const conditionalLabelProps = getConditionalLabelProps(label, isLabeled);
+
   const renderEditing = (
-    <label htmlFor={label}>
+    <>
+      {label && isLabeled && (
+        <label htmlFor={attribute || label}>
+          <span className={styles.LabelText}>{label || attribute}</span>
+        </label>
+      )}
       <Checkbox
+        {...conditionalLabelProps}
         {...rest}
         {...checkbox}
         className={styles.Input}
-        id={label}
         name={attribute || label}
         ref={formMethods?.register}
       />
       {label && isLabeled && <span className={styles.LabelText}>{label}</span>}
-    </label>
+    </>
```

Comment:
I just happened to catch this luckily. For a checkbox, the `<label>` tag is supposed to wrap the checkbox input itself. It looks like that is being changed here.